Use an int to store the size of the internal representation of a large PropertyValue.

p-f opened the pull request:
When writing a `PropertyValue` of a variable length where the size is larger than `Short.MAX_VALUE`, an `int` is used instead. In that case an additional bit-flag is set in the type byte. (`PropertyValue.FLAG_LARGE`)

Change test cases for larger properties to not expect an Exception.
Add test cases for serialization of larger properties.
Remove `validateBytesLength` method as the new maximum size is large enough to support any array. (The JVM does not support larger arrays.)

Some additional testing for larger Maps and BigDecimals may be necessary.

fixes #776

s1ck commented on gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValue.java:
```diff
@@ -117,10 +117,24 @@
+  /**
+   * Was previously used to indicate maximum size of the internal byte representation.
+   *
+   * @deprecated The new maximum size is depending on the JVM implementation.
```
Maybe hint the user to the implementation that should be used now (since it's deprecated)

s1ck commented on gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValue.java:
```diff
@@ -99,15 +99,15 @@
   public static final transient byte TYPE_LIST         = 0x0a;
   /**
-   * {@code <property-type>} for {@link java.util.List}
+   * {@code <property-type>} for {@link LocalDate}
```
thx for fixing my c&p laziness :+1:

s1ck commented on gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValue.java:
```diff
@@ -117,10 +117,24 @@
+   * stored as an {@code int} instead of a {@code short},
    */
-  public static final int MAX_BINARY_LENGTH = Short.MAX_VALUE - OFFSET;
+  public static final transient int LARGE_PROPERTY_THRESHOLD = Short.MAX_VALUE;
```
I don't like the name that much but I also don't have a better idea.

s1ck commented on gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/PropertyValue.java:
```diff
@@ -117,10 +117,24 @@
+   * @deprecated The new maximum size is depending on the JVM implementation.
+   */
+  @Deprecated
+  public static final transient int MAX_BINARY_LENGTH = Short.MAX_VALUE - OFFSET;
```
I wonder if we still need that at all. Iirc, it's only used in the validation, which is deleted now.

p-f commented:
Removed the `MAX_BINARY_LENGTH` constant, because it wasn't used anywhere.
Added link from `FLAG_LARGE` and `LARGE_PROPERTY_THRESHOLD` to `write(DataOutputView)`,
because the byte representation is documented there.
